src/WARDuino/WARDuino.cpp

```diff
@@ -988,6 +988,12 @@ void WARDuino::free_module_state(Module *m) {
     }
 
     if (m->callstack != nullptr) {
+        for (int j = 0; j <= m->csp; j++) {
+            Frame *f = &m->callstack[j];
+            if (f->block->block_type == 0xfe || f->block->block_type == 0xff) {
```

Comment:
Can we replace this with a nullptr check? And set the block to nullptr when we free in pop_block.
That way guard blocks that were not popped yet also get freed.